File: Top_Interview_150_List_With_solution(leetcode)/Array|String/L55-JumpGame.java
// https://leetcode.com/problems/jump-game/
class Solution {
    public boolean canJump(int[] nums) {
        int goal = nums.length - 1;
        for (int idx = nums.length - 1; idx > -1; idx--) {
            if (idx + nums[idx] >= goal) {
                goal = idx;
            }
        }
        return goal == 0 ? true : false;
    }
}

File: Top_Interview_150_List_With_solution(leetcode)/Array|String/L55-jump_game.py
# https://leetcode.com/problems/jump-game/
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        numLen = len(nums)
        goal = numLen - 1
        for idx in range(numLen - 1, -1, -1):
            if idx + nums[idx] >= goal:
                goal = idx
        return True if goal == 0 else False
